electrum-client/ElectrumClient.ts

```diff
@@ -605,4 +624,41 @@ export class ElectrumClient {
             tx.inputs.some(input => input.address && addresses.has(input.address))
             || tx.outputs.some(output => output.address && addresses.has(output.address)));
     }
+
+    private async callAgents<R>(call: AgentCall<R>, onAgentError?: AgentErrorCallback, stopAfterFirstResult?: true): Promise<R>;
+    private async callAgents<R>(call: AgentCall<R>, onAgentError: AgentErrorCallback | undefined, stopAfterFirstResult: false): Promise<R[]>;
+    private async callAgents<R>(call: AgentCall<R>, onAgentError?: AgentErrorCallback, stopAfterFirstResult: boolean = true): Promise<R | R[]> {
+        const untriedConnectedAgents: Set<Agent> = new Set(this.agents);
+        const results: R[] = [];
+        for (const agent of this.agents) {
+            try {
+                untriedConnectedAgents.delete(agent);
+                results.push(await call(agent));
+                if (stopAfterFirstResult) break;
+            } catch (error) {
+                if (onAgentError) {
+                    onAgentError(agent, error);
```

Comment:
I think the convention is that the `error` is the first parameter to an error callback. `agent` should be the second, in case the error handler does not use it.